modules/apps/foundation/configuration-admin/configuration-admin-action/src/main/java/com/liferay/configuration/admin/action/ConfigurationModelActionException.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.configuration.admin.action;
+
+import com.liferay.portal.kernel.exception.PortalException;
+
+/**
+ * @author Drew Brokke
+ */
+public class ConfigurationModelActionException extends PortalException {
+
+	public ConfigurationModelActionException(String msg) {
```

Comment:
Yes, you would need to call super(msg) if you want to inherit PortalException (which I guess we should do).  The pattern is similar to the new exceptions you guys worked on for 7.0 such as UserIdException. Note how there we deprecated all constructors that received and String and a Throwable. And then in the nested exceptions we introduced more meaningful params and generated the message. In this case we don't need the nested exception, but everything else is the same.

The message could be generated more or less like this:
super(String.format("Error executing listener %s after saving the configuration for %s", listenerClass, configurationPid))

Where listenerClass and configurationPid would be public fields and passed over in the constructor when the exception is thrown.

Makes sense?